From 9fc97ea817762e2575954cd31aa735b2cfec7a5e Mon Sep 17 00:00:00 2001
From: OPNA2608 <opna2608@protonmail.com>
Date: Sun, 3 Aug 2025 19:08:30 +0200
Subject: [PATCH] build/Makefile: Fix recursive make & parallelism

- Recursively calling make should be done by using the special `MAKE` variable, which will preserve parallelism across calls
- Don't hardcode parallelism in recursive make calls, inherit user-supplied ones instead
---
 build/Makefile | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/build/Makefile b/build/Makefile
index 6633ebd..807bfd9 100644
--- a/build/Makefile
+++ b/build/Makefile
@@ -2,36 +2,36 @@ UNAME := $(shell uname)
 
 all:
 ifeq ($(UNAME), Darwin)
-	make -j 5  -f Makefile.MacOS-OMP $(MFLAGS)  
+	$(MAKE) -f Makefile.MacOS-OMP $(MFLAGS)
 endif
 ifeq ($(UNAME), Linux)
-	make -j 3  -f Makefile.ubuntu $(MFLAGS)
+	$(MAKE) -f Makefile.ubuntu $(MFLAGS)
 endif
 ifeq ($(UNAME), Solaris)
-	make  -f Makefile.solaris $(MFLAGS)
+	$(MAKE) -f Makefile.solaris $(MFLAGS)
 endif
 
 install:
 ifeq ($(UNAME), Darwin)
-	make -j 5  -f Makefile.MacOS-OMP $(MFLAGS) install
+	$(MAKE) -f Makefile.MacOS-OMP $(MFLAGS) install
 endif
 ifeq ($(UNAME), Linux)
-	make -j 3  -f Makefile.ubuntu $(MFLAGS) install
+	$(MAKE) -f Makefile.ubuntu $(MFLAGS) install
 endif
 ifeq ($(UNAME), Solaris)
-	make  -f Makefile.solaris $(MFLAGS) install
+	$(MAKE) -f Makefile.solaris $(MFLAGS) install
 endif
 
 
 plugins:
 ifeq ($(UNAME), Darwin)
-	make -j 5 -f Makefile.MacOS $(MFLAGS) plugins
+	$(MAKE) -f Makefile.MacOS $(MFLAGS) plugins
 endif
 ifeq ($(UNAME), Linux)
-	make -j 3  -f Makefile.ubuntu $(MFLAGS) plugins
+	$(MAKE) -f Makefile.ubuntu $(MFLAGS) plugins
 endif
 ifeq ($(UNAME), Solaris)
-	make  -f Makefile.solaris $(MFLAGS) plugins
+	$(MAKE) -f Makefile.solaris $(MFLAGS) plugins
 endif
 
 
-- 
2.50.1

